### PR TITLE
Gamified fix logic with isComplete issue #199 2.0

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Addressing the problem where only exercise components inside the swiper had isComplete set to true has been resolved. Now, all components are updated, and the swiper starts at the first incomplete component, if any, instead of the start. Additionally, there is a refactoring of the points logic, but the underlying logic remains unchanged.

## Changes

The changes are related to issue https://github.com/Educado-App/educado-mobile/issues/199

## Related Issues

Issue #199 

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## If mobile/frontend pull request, what version of the backend is it stable, and running on?

Branch: gamified_dev_copy_nov27

Commit id: de67564

## Notes for Reviewers

I have commented out some of the tests since the functions do not exist and they can't be found in the current dev branch. Please test with a new user and the third section of the course Finance

### CODE REVIEW

**Comment by benceisonline** [comment]
all tests pass and everything works on simulator with the corresponding backend 🥳 🥳 🥳

**Review by Donhusum — CHANGES_REQUESTED** [review]
I just tried to run it with backend specified. With the user I was already logged in with, i could not se any lectures or exercises, they just load forever. I logged out and tried to log in again to see if that was the issue. I was unable to log in with an existing user or with a new user. These things needs to be fixed before this PR should be approved.

**Comment by benceisonline** [comment]
> I just tried to run it with backend specified. With the user I was already logged in with, i could not se any lectures or exercises, they just load forever. I logged out and tried to log in again to see if that was the issue. I was unable to log in with an existing user or with a new user. These things needs to be fixed before this PR should be approved.

Sounds like you are not connected to the db, did you make sure to update the IP in the api and userApi files? 
I have just tested again to make sure and everything works fine for me.

**Comment by loui981c** [comment]
try to unsub and sub to the course again

**Comment by Donhusum** [comment]
> try to unsub and sub to the course again

Hard to test since I can't log in on you setup. I tried using the dev backed instead of yours. Now I can log in no problem, but the lectures and exercises are still not showing, even when I unsub and sub again.

**Comment by benceisonline** [comment]
@Donhusum can you please make sure that you have the correct PR pulled, have the correct config files, and your own IP set in api and userApi in mobile? I cannot reproduce your errors.

File: eml/api/api.js
Comment (Jatewo) [inline eml/api/api.js:5]: Is it possible to make a .env file for the URLs? And why the staging URL when we're not trying to merge to staging?

File: eml/api/userApi.js
Comment (Jatewo) [inline eml/api/userApi.js:11]: See comment above

File: eml/api/userApi.js
Comment (Jatewo) [inline eml/api/userApi.js:128]: What is the role of `isComplete` here? Please add some comments above the functions.

File: eml/api/userApi.js
Comment (Jatewo) [inline eml/api/userApi.js:130]: Does this follow our API guidelines?

File: eml/api/userApi.js
Comment (Jatewo) [inline eml/api/userApi.js:238]: API guidelines?

**Review by PrometheusBork — APPROVED** [review]
Works fine on my device

File: eml/services/utilityFunctions.js
Comment (Jatewo) [inline eml/services/utilityFunctions.js:1]: Perhaps give this file a better name. We should not put all utility functions in one file

File: eml/test/services/StorageServices.test.js
Comment (Jatewo) [inline eml/test/services/StorageServices.test.js:1]: Why are these commented out?

**Review by sofiiagran — APPROVED** [review]
Great job! I approve BUT: do not merge before your backend is in in dev!!!!   There are some problems since the database have not been updated yet (error when loading empty sections/sections with all model design). However, this will not stop this PR to be merged to dev, and hopefully the db will be fixed asap. Also video does not work on this PR, but when your backend is merged to staging, and this PR can use staging db, this problem will be fixed.

**Review by Jatewo — CHANGES_REQUESTED** [review]
I have requested some changes. Please take a look at them 😄

File: eml/api/api.js
Comment (loui981c) [reply eml/api/api.js:5]: i think it is an integration issue and not related to this PR. this is meant to go to staging at some point, thats why its the staging url and not local

File: eml/test/services/StorageServices.test.js
Comment (loui981c) [reply eml/test/services/StorageServices.test.js:1]: this is offline's test which fails bc change of structure. i have talked to them about it

File: eml/api/userApi.js
Comment (loui981c) [reply eml/api/userApi.js:128]: isComplete is the bool to set the component to complete in backend. If u look at the exerciseScreen, u can see that if user answers wrong then isComplete is false

File: eml/test/services/StorageServices.test.js
Comment (benceisonline) [reply eml/test/services/StorageServices.test.js:1]: an issue is also made in a sec

**Pull request merged**
